force-app/main/default/classes/PerformanceExperiment.cls

```diff
@@ -10,27 +10,45 @@ public with sharing class PerformanceExperiment implements Queueable {
     private Integer stepSize;
     private Integer endSize;
     private Integer repetitions;
-    private PerformanceScenario experiment;
-    
     private Integer thisRepetition;
-    private PerformanceExperiment thenRun;
+    private Integer totalExecutions;
+    
+    private String experimentType;
+    private String serializedThenRun;
+    
+    public PerformanceScenario experiment;
+    public PerformanceExperiment thenRun;
 
-    public String suiteName {get; private set;}
-    public String experimentName {get; private set;}
-    public Integer thisStep {get; private set;}
+    public String suiteName { get; private set; }
+    public String experimentName { get; private set; }
```

Comment:
`experimentName` and the label coming from calling `getLabel()` on the scenario are the same concept. So, only one needs to exist. 

I'd say that we can get rid of `experimentName`. That field comes all the way from the suite, so it will be a nice amount of code that we can remove here.